Web/stockdogweb/src/components/Listing/Listing.js

```diff
@@ -13,55 +13,65 @@ class Listing extends Component {
 
    generateListings = () => {
       var listingElements = [];
+
+      // Dont' do anything if there are no listings
+      if (!this.props.listings) {
+         return;
+      }
+
       this.props.listings.forEach(listing => {
 
          // Need to do the priceChange element seperately because
          // it changes depending on if its negative or positive
          var priceChangeElement;
 
          if (listing.priceChange >= 0) {
-            priceChangeElement = <div class="listing-item-price-change listing-up-color">
-               (+{listing.priceChange})
-            </div>
+            priceChangeElement = 
+               <div className="listing-item-price-change listing-up-color">
+                  (+{listing.priceChange})
+               </div>
          }
          else {
-            priceChangeElement = <div class="listing-item-price-change listing-down-color">
-               ({listing.priceChange})
-            </div>
+            priceChangeElement = 
+               <div className="listing-item-price-change listing-down-color">
+                  ({listing.priceChange})
+               </div>
```

Comment:
Not super necessary but you could set a variable for the listing price change and consolidate these elements into one statement (like if listing.priceChange >= 0 set a variable priceChangeString = '+' + listing.priceChange and etc for the else statement and then after the if else make the element with the priceChangeString in it so that it's not duplicate)

Reply:
Good suggestion!